### PR TITLE
Allow descriptive error names by default in `catch-error-name`

### LINKED ISSUE #192: Allow descriptive error name in catch-error-name
Currently the `catch-error-name` forces to name the `catch` argument `error`, `error2`, `error3` etc... Adding a number at the end of a variable is not really descriptive.

I guess it's ok for the rule to accept `error${index}` as a valid name but the rule should allow and recommend to use a descriptive error name such as `fsError`, `httpError` etc...

### PR DESCRIPTION
**Pull request by lo1tuma**
Fixes #192.

### CODE REVIEW

File: rules/catch-error-name.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ const create = context => {
 
 	const options = {
 		name: 'error',
-		caughtErrorsIgnorePattern: '^_$',
+		caughtErrorsIgnorePattern: '^_$|^error\\w+|^[A-Za-z]\\w*(e|E)rror$',
```
Comment (sindresorhus) [inline rules/catch-error-name.js]:
```suggestion
		caughtErrorsIgnorePattern: /^_$|^error\w+|^[A-Za-z]\w*(e|E)rror$/.source,
```

Comment (sindresorhus) [reply rules/catch-error-name.js]: I don't think we should allow `_` in the same.

Comment (sindresorhus) [reply rules/catch-error-name.js]: This needs to be updated in the docs too.

**Comment by sindresorhus** [comment]
Maybe add a tip to the docs about the fact that even though auto-fix fixes it to `error2`, `error3`, etc, the user is free to rename it to something more descriptive as long as it matches the default pattern.

**Comment by lo1tuma** [comment]
Thanks fo the feedback. I’ve updated the PR and documented the new default setting.

> I don't think we should allow `_` in the same.

Do you mean the case where the name is just `_` or that we allow snake case like `network_error`?

**Comment by sindresorhus** [comment]
Snake case

**Comment by fisker** [comment]
Do we need allow `someError2` ?

File: rules/catch-error-name.js [hunk 2]
```diff
@@ -34,7 +34,7 @@ const create = context => {
 
 	const options = {
 		name: 'error',
-		caughtErrorsIgnorePattern: '^_$',
+		caughtErrorsIgnorePattern: /^_$|^error\w+|^[A-Za-z]\w*(e|E)rror$/.source,
```
Comment (sindresorhus) [inline rules/catch-error-name.js]: Should we really allow `errorFoo`? Seems like the more correct naming would be `fooError`.

**Comment by sindresorhus** [comment]
> Do we need allow someError2 ?

Let's try without first and see if anyone complains.

File: rules/catch-error-name.js (hunk 2 above)
Comment (sindresorhus) [reply rules/catch-error-name.js]: Are you asking "why don't we support regex **literal** directly"?

Comment (fisker) [reply rules/catch-error-name.js]: Sorry, yes. I mean we can support regex literal and regex string.

Comment (sindresorhus) [reply rules/catch-error-name.js]: How? The JSON schema does not support regex literal.

Comment (fisker) [reply rules/catch-error-name.js]: Right,never mind.

Comment (sindresorhus) [reply rules/catch-error-name.js]: Relevant: https://github.com/eslint/eslint/issues/11318

Comment (lo1tuma) [reply rules/catch-error-name.js]: Removed support for `errorFoo`.

**Comment by lo1tuma** [comment]
Removed support for snake case.

File: rules/catch-error-name.js [hunk 3]
```diff
@@ -34,7 +34,7 @@ const create = context => {
 
 	const options = {
 		name: 'error',
-		caughtErrorsIgnorePattern: '^_$',
+		caughtErrorsIgnorePattern: /^_$|^[A-Za-z0-9]+(e|E)rror$/.source,
```
Comment (fisker) [inline rules/catch-error-name.js]:
```suggestion
		caughtErrorsIgnorePattern: /^_$|^[\dA-Za-z]+(e|E)rror$/.source,
```

**Pull request merged**